onnx/shape_inference/implementation.cc

```diff
@@ -229,6 +229,8 @@ void MaterializeSymbolicShape(TypeProto* inferred_type, SymbolTable& symbol_tabl
     MaterializeSymbolicShape(inferred_type->mutable_sequence_type()->mutable_elem_type(), symbol_table);
   } else if (inferred_val_case == TypeProto::kOptionalType) {
     MaterializeSymbolicShape(inferred_type->mutable_optional_type()->mutable_elem_type(), symbol_table);
+  } else if (inferred_val_case == TypeProto::kMapType) {
```

Comment:
Thanks for this fix. This was also uncovered in an earlier PR adding more map ops, but that PR seems to be on hold. However, it does have one more line of fix which may be relevant and worth adding here: see [this PR](https://github.com/onnx/onnx/pull/4640/files#diff-df8e9d76c8ada1b583a5dd877963bd651fbf76ec0cac241d05bf97af4ef8c8f4)

Reply:
![image](https://user-images.githubusercontent.com/10075881/236048039-24a2530d-f17e-45f9-9591-2cca0c363da6.png)

Reply:
The change to `mergeShapesAndTypes` is highlighted above

Reply:
Thanks! I took a short look and it seems the only other general fix is for `from_dict` in `numpy_helper`, but I think that function should probably be rewritten in another PR.

Reply:
I'm unsure why tests didn't catch this, though. I think the code isn't actually entered in our shape inference tests, as if the type was entirely undefined before, the inferred type is copied directly, so the undefined-key-type branch was never hit: 
https://github.com/onnx/onnx/blob/ebb64fcbf48d81fe82af2c551e6ca514e553071b/onnx/shape_inference/implementation.cc#L260-L274